# ans 1->
estimation in stastistics is numerous procedure used to calculate the value of some property of a population 
        from observation of a sample drawn from the population.and estimation is a numerical value used to estimate
        unknown population parameter.
        there are two types of estimation-
    (1) point estimation - a point estimation of a population parameter is a single value used to estimate the 
                           unknown population.
    (2)interval estimation - it provide a range of values used to estimate or calculate the unknown population parameters
                             for example - the mean(average)-of a population by computing range.
                 

# ans 2-> 

In [17]:
import math

def estimate_population_mean(sample_mean, sample_std_deviation, sample_size):
    # Calculate the standard error
    standard_error = sample_std_deviation / math.sqrt(sample_size)
    
    # Calculate the lower and upper bounds of the estimate
    lower_bound = sample_mean - standard_error
    upper_bound = sample_mean + standard_error
    
    # Return the estimate as a tuple (mean, lower_bound, upper_bound)
    return sample_mean, lower_bound, upper_bound

# Example usage:
sample_mean = 65
sample_std_deviation = 8
sample_size = 50

mean_estimate, lower_bound, upper_bound = estimate_population_mean(sample_mean, sample_std_deviation, sample_size)

print(f"Estimated Population Mean: {mean_estimate}")
print(f"Confidence Interval: ({lower_bound}, {upper_bound})")

Estimated Population Mean: 65
Confidence Interval: (63.868629150101526, 66.13137084989847)


# ans 3->
Hypotesis testing- it is a form of statistical inferences that uses data from a sample to draw conclusions about 
         a population parameter or a population probability distribution.
    It is used to support scientific research and create breakthroughs in knowledge.
* Importance-->
                (1)it helps to assume the probability of research failure and progress.
                (2) it helps to provide link to the underlying theory and specific research question.
                (3)it helps in data analysis and measures the validity and reliability of the research.
    

# ans 4-


* null hypothesis - The average weight of male college students is equal to or less than the average weight of female college students.

*  alternate hypotheesis - The average weight of male college students is greater than the average weight of female college students.


# ans 6->
null hypothesis of a test always predict that there is no effect or no relation between variables andd population.
alternative hypothesis predict that there is effect on population and relation in variables.
null hypothesis and alternative hypothesis are work alternative of each other.if null hypthesis is accepted then alternative
hypothesis is rejected
for an example -   as per null hypothesis,there is no change in water quality in the first half of the year as compared to 
                  the second half.but in alternative hypothesis,the quality of water is poor in the second half when observed.

# ans 7->
some steps involve in hypostesis testing are

1. specify the hypothesis i.e. null or alternative
2. choose a significance level for the test
3. collect a sample of data and compute the test statistic
4. compute the critical value
5. compare the test statistic to the critical value
6. interpret the result and draw conclusion

# ans 8->  
P-value is defined as the measure of the probability that an observed difference might have occurred just by 
random chance. p value helps to find out how likely a particular set of observation occured.
p values used in statistical hypothesis testing to reject the null hypothesis.the smaller the p-value,the stronger 
the likelihood that you should reject the null hypothesis and greater the statistical significance of the observed difference.


# ans 9->

In [1]:
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
import pandas as pd

FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\User\\AppData\\Roaming\\Python\\Python311\\site-packages\\matplotlib.libs'

In [26]:
pip install matplotlib!

Note: you may need to restart the kernel to use updated packages.


ERROR: Invalid requirement: 'matplotlib!'


In [ ]:
firstT = stats.t.pdf(x,10)


In [19]:
plt.plot(x,firstT)

NameError: name 'plt' is not defined

# ans 10-> 
python program to calculate two sample test

In [23]:
import numpy as np
import scipy as scipy
from scipy import stats
import pandas  as pd

population1 =14, 15, 15, 16, 13, 8, 14, 17, 16, 14, 19, 20, 21, 15, 15, 16, 16, 13, 14, 12
population2 = 15, 17, 14, 17, 14, 8, 12, 19, 19, 14, 17, 22, 24, 16, 13, 16, 13, 18, 15, 13


In [22]:
## null phypthesis mean of group1 =mean of group2
##alternste hypothesis meam of group1=/ mean of group2

In [8]:
stats.ttest_ind(population1,population2,equal_var=True)

Ttest_indResult(statistic=-0.6337397070250238, pvalue=0.5300471010405257)

here the p value (0.5300) is greater than 0.05 so we fail to reject to the null hypothesis the mean of population mean is not 
equal.

# ans 11->
students's t-distribution is a hypothetical  pobability distribution.it is used to make presumptions about a mean    when standard deviation is not known to us.it is symmetrical,bell-shaped distribution,similar to standard normal curve. the student t distribution is similar to normal distriution but flatter and shorter than normal distribution.
 
  uses of student t distribution
  (1)when the standard deviation is not known  then t distribution is used
  (2)when the sample size is smal i.e.(n<30). 

# ans 12->
the test statistic is a number calculated from a statistical test of a hypothesis.
t statistic is used in a T test when you are deciding whether to support or reject the null hypothesis.it can be 
found in so many different ways.the yest statistic is used to calculate the p value.
                                           
Formula for t statistic                       
   t= observed mean or sample mean - theortical mean or population mean/standard deviation*root(n)

# ans 13-
* Sample Mean = 500
* Standard Deviation = 50
* Sample Size (n) = 50
* Critical Value (Z) = 1.96

Confidence Interval= sample mean +- Z* sd/root(n)
 = 500 +- 1.96*50/root(50)

henca this is calacaluted and we find that estimated average daily income of coeffe shop ownwer is between $486.13 and $513.87

# ans 15-
* null hpothesis - the true mean weight of the products is equal to 5 pounds.
* alternative hypothesis -  the true mean is less than 5 pounds.

* t = (sample mean - population mean)* root(sample size)/std 

sd=0.5  
population mean=5
sample size n=25
Sample mean = 4.8

t test= 4.8-5*25/0.5=-2
 
    we find significance levrl at 0.01 at 24 degree of freedom is -2.492
    
    on comparing calculted t value with sugnficance itnerval -2>-2.492 we fail to reject the null hypothsis  

# ans 16-
* n1=30,     n2=40
* std1=10,    std2=8
* mean1=80,   mean2=75
* alpa=0.01
 

In [ ]:
t_test=mean1-mean2/root()

# ans 17-
* sample size(n)=50
* sample mean=4
* std=1.5
* critical value for 99% ci(Z)= 2.576

estimated confidence interval=sample mean+- Z(srd/n)

=4 +-2.576*1.5/sqrt(50)

Estimated CI is between 3.6186 and 4.3814.